File: leetcode_problems/typescript/src/lc1662_check_if_two_strings_arrays_are_equivalent.ts
function arrayStringsAreEqual(word1: string[], word2: string[]): boolean {
  let ap1 = 0;
  let wp1 = 0;
  let ap2 = 0;
  let wp2 = 0;
  while (ap1 < word1.length && ap2 < word2.length) {
    let char1 = word1[ap1][wp1];
    let char2 = word2[ap2][wp2];
    if (char1 != char2) {
      return false;
    }

    wp1++;
    wp2++;

    if (wp1 >= word1[ap1].length) {
      wp1 = 0;
      ap1++;
    }
    if (wp2 >= word2[ap2].length) {
      wp2 = 0;
      ap2++;
    }
  }
  return ap1 == word1.length && ap2 == word2.length;
}

console.log(arrayStringsAreEqual(["ab", "c"], ["a", "bc"]));

File: leetcode_problems/cpp/lc1662_check_if_two_strings_arrays_are_equivalent.cpp
#include <vector>
#include <string>
#include <iostream>
using std::string;
using std::vector;

class Solution
{
public:
    bool arrayStringsAreEqual(vector<string> &word1, vector<string> &word2)
    {
        int ap1 = 0;
        int wp1 = 0;
        int ap2 = 0;
        int wp2 = 0;
        while (ap1 < word1.size() && ap2 < word2.size())
        {
            char chr1 = word1[ap1][wp1];
            char chr2 = word2[ap2][wp2];
            if (chr1 != chr2)
            {
                std::cout << "exit " << chr1 << " " << chr2 << std::endl;
                return false;
            }
            wp1++;
            wp2++;

            if (wp1 >= word1[ap1].size())
            {
                ap1++;
                wp1 = 0;
            }

            if (wp2 >= word2[ap2].size())
            {
                ap2++;
                wp2 = 0;
            }
        }
        return ap1 == word1.size() && ap2 == word2.size();
    }
};

int main()
{
    vector<string> word1 = {"ab", "c"};
    vector<string> word2 = {"a", "bc"};
    Solution sol;
    std::cout << sol.arrayStringsAreEqual(word1, word2) << std::endl;
    return 0;
}